src/context.cpp

```diff
@@ -96,6 +104,25 @@ const memory::unordered_map<rule *, filter_mode> &context::filter_rules(ddwaf::t
     return rules_to_exclude_;
 }
 
+void context::eval_preprocessors(optional_ref<ddwaf_object> &derived, ddwaf::timer &deadline)
+{
+    for (const auto &[id, preproc] : ruleset_->preprocessors) {
+        if (deadline.expired()) {
+            DDWAF_INFO("Ran out of time while evaluating preprocessors");
```

Comment:
NP: I'm doubtful about this being a `INFO`

Reply:
Timeouts are relatively common so info is a reasonable compromise to avoid spamming customers only interested in warning / error logs. The library can decide to surface such warnings once they inspect `ddwaf_result`.

Reply:
Out of interest, why are they common? I thought it would be more rare

Reply:
It depends, the timeout is configurable and in some cases (e.g. gRPC), the default timeout used is very low.